cmd/thanos/config.go

```diff
@@ -74,22 +75,13 @@ func (pc *prometheusConfig) registerFlag(cmd extkingpin.FlagClause) *prometheusC
 	cmd.Flag("prometheus.ready_timeout",
 		"Maximum time to wait for the Prometheus instance to start up").
 		Default("10m").DurationVar(&pc.readyTimeout)
-	return pc
-}
+	pc.httpClient = extflag.RegisterPathOrContent(
+		cmd,
+		"prometheus.http-client",
+		"YAML file or string with http client configs. see Format details : ...",
```

Comment:
@someshkoli One small nit. I think we can remove `see Format details...`.

Reply:
Yep, its redundant

Reply:
Should I make another PR for this ? 😂😂😂

Reply:
Feel free to.

Reply:
It would be good to include it in a pr for auth documentation. Would you like to take it?

Reply:
Yeah, I'll make one 🙌

Reply:
It would be great if we can have document for this feature.